### PR TITLE
Fix LIME example numpy - pandas conversion

### PR DESCRIPTION
**Pull request by FrancescMartiEscofetQC**
This PR fixes an error in the LIME example in case the `SLearner` was used. Inside the `SLearner` we convert to numpy arrays to pandas if the base model supports categoricals variables, this raised an error as the categorical variables were not set properly (at train there was the treatment and other categorical variables and at prediction only the treatment was categorical).
For this I changed the tutorial to handle this case. Now instead of manually encoding the categorical codes in the numpy array, in the modified `predict` function we reconvert to a dataframe with the correct categorical variables.

[Rendered version](https://metalearners--57.org.readthedocs.build/en/57/examples/example_lime.html)

# Checklist

- [ ] Added a `CHANGELOG.rst` entry

### CODE REVIEW

File: docs/examples/example_lime.ipynb [hunk 1]
```diff
@@ -244,7 +248,11 @@
     "from metalearners.utils import simplify_output\n",
     "\n",
     "def predict(X):\n",
-    "    return simplify_output(rlearner.predict(X, is_oos=True))"
+    "    X_pd = pd.DataFrame(X, copy=True)\n",
+    "    for c in X_pd.columns:\n",
+    "        # This line sets the cat.categories correctly (even if not all are present in X)\n",
+    "        X_pd[c] = X_pd[c].astype(df[feature_columns].iloc[:, c].dtype)\n",
```
Comment (kklein) [inline docs/examples/example_lime.ipynb:254]:
```suggestion
    "        X_pd[c] = X_pd[c].astype(df[c].dtype)\n",
```
Would this work, too?

**Review by kklein — COMMENTED** [review]
Great catch!

File: docs/examples/example_lime.ipynb (hunk 1 above)
Comment (FrancescMartiEscofetQC) [reply docs/examples/example_lime.ipynb:254]: No, the issue here is that the numpy array does not have column names so when converting it to a `dataframe` the columns names are `0,1,2...` so we need to use `iloc` to access the columns in the original `dataframe`.

Comment (kklein) [reply docs/examples/example_lime.ipynb:254]: That makes perfect sense - thanks for explaining!

**Review by kklein — APPROVED** [review]
Thanks! :)

**Pull request merged**